### ISSUE TITLE
Nameko crashes when consuming illegal messages

### ISSUE THREAD

**Issue by xiexiaopeng2016** [nameko/nameko#753]
File "/home/xiexiaopeng/.local/lib/python3.6/site-packages/nameko/rpc.py", line 118, in handle_message
    self.handle_result(message, None, exc_info)
  File "/home/xiexiaopeng/.local/lib/python3.6/site-packages/nameko/rpc.py", line 133, in handle_result
    result, exc_info = responder.send_response(result, exc_info)
  File "/home/xiexiaopeng/.local/lib/python3.6/site-packages/nameko/rpc.py", line 219, in send_response
    routing_key = self.message.properties['reply_to']
KeyError: 'reply_to'

  File "/home/xiexiaopeng/.local/lib/python3.6/site-packages/nameko/rpc.py", line 118, in handle_message
    self.handle_result(message, None, exc_info)
  File "/home/xiexiaopeng/.local/lib/python3.6/site-packages/nameko/rpc.py", line 133, in handle_result
    result, exc_info = responder.send_response(result, exc_info)
  File "/home/xiexiaopeng/.local/lib/python3.6/site-packages/nameko/rpc.py", line 201, in send_response
    content_type = self.message.properties['content_type']
KeyError: 'content_type'

**Comment by xiexiaopeng2016** [comment]
@mattbennett

**Comment by mattbennett** [comment]
This is a bug, thanks for reporting.

The RPC responder has logic on the very next lines to send a generic response in case the result cannot be serialised. We should expand that logic to also handle the case where the message itself is illegal.

See https://github.com/nameko/nameko/blob/6c7c959bdcf2f94736de6565ff4c54d6563a423c/nameko/rpc.py#L238-L253

Out of curiosity, which client are you using when this happens?

**Comment by vmesel** [comment]
Hey @mattbennett, I'm using nameko and as well receiving the same error. I'm sending messages through RabbitMQ's web interface.

**Comment by mattbennett** [comment]
Thanks for sharing @vmesel. If you're sending messages through the RabbitMQ's web interface you will need to set the `content_type` property. This bug will mean that your Nameko service will crash if you don't, but even after it's fixed you'd still get a serialisation error if that property isn't set.

**Comment by vmesel** [comment]
Yes, I tested and noticed that yesterday before this issue. The issue that I’m facing right now is how I can see if the message was processed or not. Whenever I send a valid JSON message with the proper header, the message is not processed. Should I open other issue for this topic?

**Comment by mattbennett** [comment]
@vmesel yes feel free to open a new issue for that topic, but I think the likely situation is that the message is not being routed to a queue. If there are no matching routes, the broker will drop the message, and it won't inform the publisher unless you have [publisher confirms](https://www.rabbitmq.com/confirms.html#publisher-confirms) enabled.
